src/adapter/src/catalog.rs

```diff
@@ -6695,20 +6700,12 @@ impl SessionCatalog for ConnCatalog<'_> {
         self.state.aws_privatelink_availability_zones.clone()
     }
 
-    fn get_feature(&self, name: CatalogFeature) -> bool {
-        use CatalogFeature::*;
-        let config = &self.state.system_configuration;
-        match name {
-            EnableWithMutuallyRecursive => config.enable_with_mutually_recursive(),
-        }
+    fn system_vars(&self) -> &SystemVars {
+        &self.state.system_configuration
     }
 
-    fn set_feature(&mut self, name: CatalogFeature, value: bool) -> bool {
-        use CatalogFeature::*;
-        let config = &mut self.state.to_mut().system_configuration;
-        match name {
-            EnableWithMutuallyRecursive => config.set_enable_with_mutually_recursive(value),
-        }
+    fn mut_system_vars(&mut self) -> &mut SystemVars {
```

Comment:
I believe the idiomatic way of doing this is have `mut` as a suffix instead of a prefix. For example, see[ `get`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get) and [`get_mut`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get_mut) in `Vec`s.